src/com/owncloud/android/operations/UploadFileOperation.java

```diff
@@ -372,6 +372,9 @@ protected RemoteOperationResult run(OwnCloudClient client) {
             if (temporalFile != null && !originalFile.equals(temporalFile)) {
                 temporalFile.delete();
             }
+            if (result == null){
+                return new RemoteOperationResult(false, 404, null);
```

Comment:
Mmmm... in theory, we should grant that ```result``` is never null at that point... that' s why it's not being checked.